src/test/scala/cognite/spark/v1/FilesRelationTest.scala

```diff
@@ -39,74 +55,300 @@ class FilesRelationTest extends FlatSpec with Matchers with SparkTest {
-      .format("cognite.spark.v1")
-      .option("apiKey", writeApiKey)
-      .option("type", "files")
-      .load()
-    df.createOrReplaceTempView("filesSource")
+    try {
+      spark
+        .sql(s"""
+                |select "name-$source" as name,
+                |null as id,
+                |'$source' as source,
+                |'externalId-$source' as externalId,
+                |null as mimeType,
+                |null as metadata,
+                |null as assetIds,
+                |null as datasetId,
+                |null as sourceCreatedTime,
+                |null as sourceModifiedTime,
+                |null as securityCategories,
+                |null as uploaded,
+                |null as createdTime,
+                |null as lastUpdatedTime,
+                |null as uploadedTime,
+                |null as uploadUrl
+     """.stripMargin)
+        .select(sourceDf.columns.map(col): _*)
+        .write
+        .insertInto("destinationFiles")
 
-    spark
-      .sql(s"""
-            |select id,
-            |name,
-            |source,
-            |id as externalId,
-            |mimetype,
-            |metadata,
-            |Array(1739842716040355) as assetIds,
-            |uploaded,
-            |uploadedTime,
-            |createdTime,
-            |lastUpdatedTime,
-            |uploadUrl,
-            |dataSetId
-            |from filesSource
-            |where source = '$originalSource'
+      val rows = retryWhile[Array[Row]](
+        spark.sql(s"select * from destinationFiles where source = '$source'").collect,
+        rows => rows.length < 1)
+      assert(rows.length == 1)
+    } finally {
+      try {
+        cleanupFiles(source)
+      } catch {
+        case NonFatal(_) => // ignore
+      }
+    }
+  }
+
+  it should "support updates using id and externalId" taggedAs WriteTest in {
+    val source = s"update-${shortRandomString()}"
+    val metricsPrefix = "updates.files"
+
+    try {
+      val destinationDf: DataFrame = spark.read
+        .format("cognite.spark.v1")
+        .option("apiKey", writeApiKey)
+        .option("type", "files")
+        .option("collectMetrics", "true")
+        .option("metricsPrefix", metricsPrefix)
+        .load()
+      destinationDf.createOrReplaceTempView("destinationFilesUpdate")
+
+      spark
+        .sql(s"""
+                |select "name-$source" as name,
+                |null as id,
+                |'$source' as source,
+                |'externalId-$source' as externalId,
+                |null as mimeType,
+                |null as metadata,
+                |null as assetIds,
+                |null as datasetId,
+                |null as sourceCreatedTime,
+                |null as sourceModifiedTime,
+                |null as securityCategories,
+                |null as uploaded,
+                |null as createdTime,
+                |null as lastUpdatedTime,
+                |null as uploadedTime,
+                |null as uploadUrl
```

Comment:
Can just use `write...save()` instead, then we won't have to specify nulls like this (we've already tested insertInto above).

Reply:
Yeah will do that !